subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -87,7 +87,16 @@ struct bt_bap_scan_delegator_inst {
 		[CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT];
 };
 
-static bool conn_cb_registered;
+enum scan_delegator_registered_flags {
+	SCAN_DELEGATOR_REGISTERED_CONN_CB,
+	SCAN_DELEGATOR_REGISTERED_SCAN_DELIGATOR,
+	SCAN_DELEGATOR_REGISTERED_PA_SYNC_CB,
+
+	SCAN_DELEGATOR_REGISTERED_FLAGS_NUM,
+};
+
+static ATOMIC_DEFINE(registered_flags, SCAN_DELEGATOR_REGISTERED_FLAGS_NUM);
```

Comment:
```suggestion
enum scan_delegator_flag {
	SCAN_DELEGATOR_FLAG_REGISTERED_CONN_CB,
	SCAN_DELEGATOR_FLAG_REGISTERED_SCAN_DELIGATOR,
	SCAN_DELEGATOR_FLAG_REGISTERED_PA_SYNC_CB,

	SCAN_DELEGATOR_FLAG_NUM,
};

static ATOMIC_DEFINE(flags, SCAN_DELEGATOR_FLAG_NUM);
```

Might have missed my suggestion in previous review: Suggest to keep the enum name more generic so that it can be expanded to non-register flags as well :)

Reply:
uh, ok ;)